app/actions.js

```diff
@@ -360,30 +367,42 @@ export const hideEmojiPicker = () => dispatch => {
   dispatch({ type: 'HIDE_EMOJI_PICKER' })
 }
 
+const getCabalUnreadMessagesCount = (cabalDetails) => {
+  const cabalCore = client._keyToCabal[cabalDetails.key]
+  const channelMessagesUnread = {}
+  cabalDetails.getChannels().map((channel) => {
+    channelMessagesUnread[channel] = client.getNumberUnreadMessages(channel, cabalCore)
+  })
+  return channelMessagesUnread
+}
+
 const initializeCabal = ({ addr, username, dispatch, settings }) => async (dispatch, getState) => {
-  const cabal = addr ? await client.addCabal(addr) : await client.createCabal()
+  const cabalDetails = addr ? await client.addCabal(addr) : await client.createCabal()
   // if creating a new cabal, addr will be undefined.
-  const { key: cabalKey } = cabal
+  const { key: cabalKey } = cabalDetails
   let firstUpdate = true
-  cabal.on('update', (details) => {
+
+  cabalDetails.on('update', throttle((details) => {
     const users = details.getUsers()
     const username = details.getLocalName()
     const channels = details.getChannels()
     const channelsJoined = details.getJoinedChannels()
+    const channelMessagesUnread = getCabalUnreadMessagesCount(details)
     const currentChannel = details.getCurrentChannel()
     const channelMembers = details.getChannelMembers()
-    dispatch({ type: 'UPDATE_CABAL', addr: cabalKey, users, username, channels, channelsJoined, currentChannel, channelMembers })
-    dispatch(getMessages({ addr: cabalKey, amount: 100, channel: currentChannel }))
+    dispatch({ type: 'UPDATE_CABAL', addr: cabalKey, channelMessagesUnread, users, username, channels, channelsJoined, currentChannel, channelMembers })
+    dispatch(getMessages({ addr: cabalKey, amount: 1000, channel: currentChannel }))
+    dispatch(updateAllsChannelsUnreadCount({ addr, channelMessagesUnread }))
     if (firstUpdate) {
       firstUpdate = false
       dispatch(viewCabal({ addr: cabalKey, currentChannel: settings.currentChannel }))
       // Focus default or last channel viewed
       dispatch(viewChannel({ addr: cabalKey, channel: settings.currentChannel }))
-    }
-  })
 
-  settings = settings || getState().cabalSettings[addr] || {}
-  dispatch(updateCabalSettings({ addr, settings }))
+      settings = settings || getState().cabalSettings[addr] || {}
+      dispatch(updateCabalSettings({ addr, settings, channelMessagesUnread }))
+    }
+  }, 2000))
```

Comment:
do you think 2 sec is a required? won't that put that much delay between messages?

Reply:
oh yeah! i started with an extreme delay for testing and thought we could work our way down to the lowest delay that we could stand. what do you think?

Reply:
I agree. Do you think we can use [ready](https://github.com/cabal-club/cabal-core/#cabalreadycb) listener? since this is only  problem initially when there are so many updates!

Reply:
yeah! although i'd like it if it were exposed through cabal-client instead. i think

Reply:
ah. okay!